### PR TITLE
Netlify previews with `--panicOnWarning`

### LINKED ISSUE #34357: Build Netlify previews with --panicOnWarning
**This is a Feature Request**

**What would you like to be added**
Update the configuration for Netlify so that we build site previews with Hugo's `--panicOnWarning` option enabled.

**Why is this needed**
We can use these warnings (and the panics they trigger) to tell us if we are introducing any problem behavior.

**Comments**
We would need to clear any existing warnings before we made this change.

### PR DESCRIPTION
**Pull request by Rishit-dagli**
Netlify preview builds now take place with `--panicOnWarning` feature of Hugo. While doing preview builds now, a `.Path` deprecation warning resulted in the error due to `--panicOnWarning` so I also fixed that in this PR. @sftim talks about why this flag is needed in #34357

Fixes #34357

/sig docs
/area web-development

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -32,7 +32,7 @@ build: module-check ## Build site with non-production settings and put deliverab
 	hugo --minify --environment development
 
 build-preview: module-check ## Build site with drafts and future posts enabled
-	hugo --buildDrafts --buildFuture --environment preview
+	hugo --buildDrafts --buildFuture --environment preview --panicOnWarning
```
Comment (tengqm) [inline Makefile:35]: no. I don't think we should go this far. It is too strict. Not all warnings are critical.

**Comment by sftim** [comment]
This isn't really a _web development_ issue (that's stuff like CSS, HTML, design). However I agree it's close.
/remove-area web-development

File: Makefile (hunk 1 above)
Comment (sftim) [reply Makefile:35]: @tengqm this PR is for a not-yet-accepted issue. Let's move the discussion [there](https://github.com/kubernetes/website/issues/34357).

File: layouts/partials/page-meta-links.html [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {{/* template adapted from Docsy theme */}}
-{{ if .Path }}
-  {{ $pathFormatted := replace .Path "\\" "/" }}
+{{ if .File.Path }}
+  {{ $pathFormatted := replace .File.Path "\\" "/" }}
```
Comment (sftim) [inline layouts/partials/page-meta-links.html:3]: I'd swap the commit order: fix this first.

@Rishit-dagli this change on its own might be OK to merge in a separate PR; what do you think?

Comment (Rishit-dagli) [reply layouts/partials/page-meta-links.html:3]: Thanks for pointing out that this commit is similar to https://github.com/kubernetes/website/pull/34360#issuecomment-1159031205 , I didn't notice this: my motivation to make this change was a failing build after https://github.com/kubernetes/website/pull/34361/commits/deacef302026e327cda4290ab14e2e3a70e61fdb

Now that #34360 is merged should I revert https://github.com/kubernetes/website/pull/34361/commits/0d732354e5ee53e9e6c5dcd0f0140ee0fad09f82 ?

Comment (Rishit-dagli) [reply layouts/partials/page-meta-links.html:3]: I believe deleting https://github.com/kubernetes/website/commit/0d732354e5ee53e9e6c5dcd0f0140ee0fad09f82 and force pushing might make more sense since #34360 will be merged before this

**Comment by Sea-n** [comment]
Refer to discussion in #34357
> Instead of panicking on warning, maybe we can report the presence of a warning by a GitHub PR status that doesn't block a merge. That's a different feature request.

I think we would close this PR for the moment, @Rishit-dagli how do you think?

**Comment by sftim** [comment]
The linked issue is closed
/close
